src/py_sonar_scanner/configuration.py

```diff
@@ -83,6 +90,23 @@ def _extract_sonar_properties(self, toml_properties):
         sonar_properties = tool_data["sonar"]
         return sonar_properties if isinstance(sonar_properties, dict) else {}
 
+    def _extract_common_properties(self, toml_properties):
+        properties = {}
+        if "tool" in toml_properties.keys() and "poetry" in toml_properties["tool"]:
```

Comment:
If we decide to move forward with this approach, I'd create tickets to cover other tools (like Hatch) and reframe this one as a Poetry-centered one as a baby step.